source/agora/consensus/protocol/Nominator.d

```diff
@@ -529,8 +529,14 @@ extern(D):
             const Block prev_block = blocks.front;
             Hash random_seed = this.ledger.getExternalizedRandomSeed(
                 this.ledger.getBlockHeight(), con_data.missing_validators);
-            const Block proposed_block = makeNewBlock(prev_block, con_data.tx_set,
-                con_data.timestamp, con_data.enrolls, random_seed, con_data.missing_validators);
+
+            Transaction[] received_tx_set;
+            if (this.ledger.getValidTXSet(con_data, received_tx_set))
```

Comment:
So it returns `true` if it fails?

Reply:
Ah wait it returns a string.

I thought this behavior was deprecated.

Reply:
This would make it clearer:

`if (auto error = this.ledger.getValidTXSet(con_data, received_tx_set))`

Although you could argue that the string is unused anyway.

Reply:
Yeah, lets just make it explicit.

Reply:
If the string is not used then it may as well be a bool. It is not an `isInvalid...` function so it is not obvious what is success status.